x-pack/plugins/infra/public/components/logging/log_text_stream/log_entry_row.tsx

```diff
@@ -50,14 +50,24 @@ export const LogEntryRow = memo(
     wrap,
   }: LogEntryRowProps) => {
     const [isHovered, setIsHovered] = useState(false);
+    const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+    const openMenu = useCallback(() => {
+      setIsMenuOpen(true);
+    }, []);
+
+    const closeMenu = useCallback(() => {
+      setIsMenuOpen(false);
+    }, []);
 
     const setItemIsHovered = useCallback(() => {
       setIsHovered(true);
     }, []);
 
     const setItemIsNotHovered = useCallback(() => {
       setIsHovered(false);
-    }, []);
+      closeMenu();
```

Comment:
It might be a consequence of the context menu being rather small at the moment, but I found I easily accidentally hover over another while trying to reach the link in the menu. Have you considered not implicitly closing the menu?

Reply:
Even if I don't close the menu it still disappears when the mouse goes out. The menu is controlled by the popover component, and this doesn't render when the mouse goes out.

If I don't mark the menu as closed it will reappear again when the user hovers that row again, which looks weird.

I think the ideal scenario is that the popover stays re-rendered even if the cursor moves away (so the menu can stay open). To make it work well a click on a different row should close any open menu. That means that the state of the menu open cannot live in the rows and needs to be moved up one level.

I could make a generic context/provider pair. An rough example could look like this:

```ts
// Wrap this with `constate` and add a provider to `page_providers.tsx`
const useContextualMenus = function() {
  const [visibleMenu, setVisibleMenu] = usState<string | undefined>();
  
  const openMenu = useCallback((menuId: string) => setVisibleMenu(menuId), []) 
  const closeMenu = useCallback(() => setVisibleMenu(undefined), []);
  const isMenuOpen = useCallback((menuId: string) => menuId === visibleMenu, [visibleMenu])
  
  return { openMenu, closeMenu, isMenuOpen };
}

const LogEntryActionsColumn = function({ logEntry, ...}) {
  const { openMenu, closeMenu, isOpen } = useContextualMenus();
  const shouldRenderMenu = isMenuOpen(logEntry.id);

  const button = <button onClick={() => { openMenu(logEntry.id) }}>
  // ...
}
```

Reply:
Yes, that should work. Maybe a more specific name like `useLogEntryActionMenu` would help a future reader to narrow down the purpose of that provider?

As to the location of the provider, should we put it into the page providers or into the scrollable stream view to restrict it to a more local scope and prevent leakage of concepts?

Reply:
Turns out none of that is necessary because the `EuiPopover`s close themselves when you click somewhere else. Yay!

Reply:
nice :tada: